Microsoft.Toolkit.Uwp/Extensions/StringExtensions.cs

```diff
@@ -0,0 +1,55 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Net;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Windows.ApplicationModel.Resources;
+
+namespace Microsoft.Toolkit.Uwp.Extensions
+{
+    /// <summary>
+    /// UWP specific helpers for working with strings.
+    /// </summary>
+    public static class StringExtensions
+    {
+        private static ResourceLoader _independentLoader = ResourceLoader.GetForViewIndependentUse();
+
+        /// <summary>
+        /// Retrieves the provided resource for the current view context.
+        /// </summary>
+        /// <param name="resourceKey">Resource key to retrieve.</param>
+        /// <returns>string value for given resource or empty string if not found.</returns>
+        public static string GetViewLocalized(this string resourceKey)
+        {
+            return ResourceLoader.GetForCurrentView().GetString(resourceKey);
+        }
+
+        /// <summary>
+        /// Retrieves the provided resource for the given key for use independent of the UI thread.
+        /// </summary>
+        /// <param name="resourceKey">Resource key to retrieve.</param>
+        /// <returns>string value for given resource or empty string if not found.</returns>
+        public static string GetLocalized(this string resourceKey)
+        {
+            return _independentLoader.GetString(resourceKey);
+        }
+
+        /// <summary>
+        /// Retrieves the provided resource for the given key for use independent of the UI thread.
+        /// </summary>
+        /// <param name="resourceKey">Resource key to retrieve.</param>
+        /// <param name="resourcePath">Resource path to retrieve.</param>
+        /// <returns>string value for given resource or empty string if not found.</returns>
+        public static string GetLocalized(this string resourceKey, string resourcePath)
+        {
+            _independentLoader = ResourceLoader.GetForViewIndependentUse(resourcePath);
```

Comment:
Don't overwrite the cached global resoucemap loader here, just construct it as needed:
```suggestion
            return ResourceLoader.GetForViewIndependentUse(resourcePath).GetString(resourceKey);
```